### PR TITLE
fixes issue #2

### PR DESCRIPTION
**Pull request by bitwave**
Merges all cves together for each package, uses the newest version for printing.

### CODE REVIEW

**Comment by ilpianista** [comment]
Great! Thanks for this PR!
Please try to polish it a bit by creating functions to be re-used.

**Comment by bitwave** [comment]
which functions exactly? ege get_cves_for_packege?

**Comment by ilpianista** [comment]
Yes. I know, my code is quite confusing too :-)

**Comment by bitwave** [comment]
i moved the code into functions, the names are under request for comments :D

**Comment by ilpianista** [comment]
Please merge all commits into one.

**Comment by bitwave** [comment]
squashed together into one commit

File: src/main.rs [hunk 1]
```diff
@@ -133,27 +134,83 @@ fn main() {
...
-                    };
-                };
-            },
-            Err(_) => { debug!("Package {} not installed", pkg) },
+    	let pkg_cves = get_asas_for_package_by_version(&pacman, &pkg, &cves);
+        
+        let (cves,version) = cves_with_relevant_version(pkg_cves, &pacman);
+        
+        if !cves.is_empty() {
+
```
Comment (ilpianista) [inline src/main.rs]: Please remove new line

File: src/main.rs [hunk 2]
```diff
@@ -133,27 +134,83 @@ fn main() {
...
-                            };
-                        },
-                        None => { print_asa(&options, &pkg, cve.cve, None) },
-                    };
-                };
-            },
-            Err(_) => { debug!("Package {} not installed", pkg) },
+    	let pkg_cves = get_asas_for_package_by_version(&pacman, &pkg, &cves);
+        
+        let (cves,version) = cves_with_relevant_version(pkg_cves, &pacman);
```
Comment (ilpianista) [inline src/main.rs]: Please use a different name instead of `cves`.

File: src/main.rs [hunk 3]
```diff
@@ -133,27 +134,83 @@ fn main() {
...
+	match pacman.query_package_version(pkg.clone()) {
+        Ok(v) => {
+            info!("Found installed version {} for package {}", v, pkg);
+            for cve in cves {
+                match cve.version {
+                    Some(ref version) => {
+                        info!("Comparing with fixed version {}", version);
+                        match pacman.vercmp(v.clone(), version.clone()).unwrap() {
+                            Ordering::Less => { 
+                                if !pkg_cves.contains_key(&Some(v.clone())) {
```
Comment (ilpianista) [inline src/main.rs]: Could you rewrite this to match pkg_cves.get_mut() return value instead to use a if statement?

File: src/main.rs [hunk 4]
```diff
@@ -133,27 +134,83 @@ fn main() {
...
+	                        		pkg_cves.insert(Some(v.clone()),vec![cve.clone()]);
+	                        	} else {
+	                        		pkg_cves.get_mut(&Some(v.clone())).unwrap().push(cve.clone());
+	                        	}
+                            },
+                            _ => {},
+                        };
+                    },
+                    None => {
+                    	if !pkg_cves.contains_key(&None) {
```
Comment (ilpianista) [inline src/main.rs]: Could you rewrite this to match pkg_cves.get_mut() return value instead to use a if statement?

File: src/main.rs [hunk 5]
```diff
@@ -133,27 +134,83 @@ fn main() {
...
-                                Ordering::Less => { print_asa(&options, &pkg, cve.cve, Some(version) ) },
-                                _ => {},
-                            };
-                        },
-                        None => { print_asa(&options, &pkg, cve.cve, None) },
-                    };
-                };
-            },
-            Err(_) => { debug!("Package {} not installed", pkg) },
+    	let pkg_cves = get_asas_for_package_by_version(&pacman, &pkg, &cves);
```
Comment (ilpianista) [inline src/main.rs]: Please fix indentation

**Comment by bitwave** [comment]
added changes to fit your request, after you give your ok i will squash all commits together into one

**Comment by ilpianista** [comment]
Looks good. Please add spaces after comma and squash everything into one commit.

**Comment by bitwave** [comment]
spaces after comma? like this "Some(v) => foobar(), "

File: src/main.rs [hunk 6]
```diff
@@ -134,81 +134,82 @@ fn main() {
 
     let pacman = alpm::Alpm::new().unwrap();
     for (pkg, cves) in infos {
-    	let pkg_cves = get_asas_for_package_by_version(&pacman, &pkg, &cves);
+        let pkg_cves = get_asas_for_package_by_version(&pacman, &pkg, &cves);
         
-        let (cves,version) = cves_with_relevant_version(pkg_cves, &pacman);
+        let (relevant_cves,version) = cves_with_relevant_version(pkg_cves, &pacman);
```
Comment (ilpianista) [inline src/main.rs]: please add space after comma (`relevant_cves, version`)

File: src/main.rs [hunk 7]
```diff
@@ -134,81 +134,82 @@ fn main() {
...
-	                        	} else {
-	                        		pkg_cves.get_mut(&Some(v.clone())).unwrap().push(cve.clone());
-	                        	}
+                            Ordering::Less => {
+                                let mut vec = match pkg_cves.get_mut(&Some(v.clone())) {
+                                    Some(v) => v.clone(),
+                                    None => Vec::new(),
+                                };
+                                vec.push(cve.clone());
+                                pkg_cves.insert(Some(v.clone()),vec);
```
Comment (ilpianista) [inline src/main.rs]: please add space after comma

File: src/main.rs [hunk 8]
```diff
@@ -134,81 +134,82 @@ fn main() {
...
-                    	} else {
-                    		pkg_cves.get_mut(&None).unwrap().push(cve.clone());
-                    	}
-					},
+                        let mut vec = match pkg_cves.get_mut(&None) {
+                            Some(v) => v.clone(),
+                            None => Vec::new(),
+                        };
+                        vec.push(cve.clone());
+                        pkg_cves.insert(None,vec);
```
Comment (ilpianista) [inline src/main.rs]: please add space after comma

**Comment by bitwave** [comment]
i added the spaces, sorry for the question, i got a bit confused :)

**Comment by ilpianista** [comment]
So, I tested this, but I got a wrong output:
```
Package openssl is affected by ["CVE-2016-2177", "CVE-2016-2178", "CVE-2016-2179", "CVE-2016-2180", "CVE-2016-2181", "CVE-2016-2182", "CVE-2016-2183", "CVE-2016-6302", "CVE-2016-6303", "CVE-2016-6304", "CVE-2016-6306"]. Update to 1.0.2.h-1!
```
`1.0.2.h-1` is the version installed on my system, while `1.0.2.i-1` fixes this.

**Comment by bitwave** [comment]
fixed + added a test

**Comment by bitwave** [comment]
fixed + added test

**Pull request merged**